src/CoverCrypt.tsx

```diff
@@ -277,75 +457,145 @@ const CoverCrypt: React.FC<{ kmsToken: string }> = ({ kmsToken }) => {
+                <Flex gap="2" direction="row">
+                  <CheckCircleIcon color="green.500" />
+                  Located key(s)' Id:{" "}
+                  <Stack>
+                    {locatedKeys.map((key, i) => (
+                      <Code key={i}>{key}</Code>
+                    ))}
+                  </Stack>
+                </Flex>
+              )}
+
               {/* ENCRYPT/DECRYPT IN BROWSER */}
-              <ButtonGroup isAttached variant="outline" isDisabled={keysId == null}>
+              <HeadingWithCode heading="Encrypt & Decrypt data in browser" />
+              <Code>/src/actions/encryptDataLocally.ts</Code>
+              <CodeHigligter codeInput={code?.encryptDataLocally} />
+              <Code>/src/actions/decryptDataLocally.ts</Code>
+              <CodeHigligter codeInput={code?.decryptDataLocally} />
+              <ButtonGroup isAttached variant="outline" isDisabled={keyPair == null}>
                 <Button onClick={() => handleEncrypt({ browser: true })} width="50%">
                   Encrypt data in browser
                 </Button>
-                <Button onClick={() => handleDecrypt({ browser: true })} width="50%" isDisabled={localEncryptedData == null}>
+                <Button onClick={() => handleDecryptLocally()} width="50%" isDisabled={localEncryptedData == null}>
                   Decrypt in browser
                 </Button>
               </ButtonGroup>
               {localEncryptedData && localEncryptedData.length > 0 && (
                 <EncryptedTable caption={"Encrypted in browser"} data={localEncryptedData} />
               )}
-              {localClearData && (
-                <>
-                  <Center gap="2">
-                    <CheckCircleIcon color="green.500" />
-                    Decrypted with access policy: <Code>{ACCESS_POLICY}</Code>
-                  </Center>
-                  <EmployeeTable caption={"Decrypted in browser"} data={localClearData} />
-                </>
-              )}
+              {localClearData && <EmployeeTable caption={"Decrypted in browser"} data={localClearData} />}
 
               {/* ENCRYPT/DECRYPT IN KMS */}
-              <ButtonGroup isAttached variant="outline" isDisabled={keysId == null}>
+              <HeadingWithCode heading="Encrypt & Decrypt data in KMS" />
+              <Code>/src/actions/encryptDataInKms.ts</Code>
+              <CodeHigligter codeInput={code?.encryptDataInKms} />
+              <Code>/src/actions/decryptDataLocally.ts</Code>
+              <CodeHigligter codeInput={code?.decryptDataLocally} />
+              <ButtonGroup isAttached variant="outline" isDisabled={keyPair == null}>
                 <Button onClick={() => handleEncrypt({ browser: false })} width="50%">
                   Encrypt data in KMS
                 </Button>
-                <Button onClick={() => handleDecrypt({ browser: false })} width="50%" isDisabled={kmsEncryptedData == null}>
```

Comment:
I can see you removed this refacto - why ?

Reply:
because it was messy, 2 differents actions in same function